backend/Utils/Edna.Utils.Http/Edna.Utils.Http/HttpClaimsExtension.cs

```diff
@@ -9,38 +9,51 @@ namespace Edna.Utils.Http
     public static class HttpClaimsExtension
     {
         private static readonly string[] PossibleEmailClaimTypes = { "email", "upn", "unique_name" };
+
         public static bool TryGetUserEmails(this IHeaderDictionary headers, out List<string> userEmails, Action<string> logAction = null)
         {
             userEmails = new List<string>();
-
             if (!headers.TryGetTokenClaims(out Claim[] claims, logAction))
             {
                 logAction("Error in sent JWT");
                 return false;
             }
 
+            return CheckCallValidity(claims, out userEmails, logAction);
+        }
+
+        private static bool CheckCallValidity(Claim[] claims, out List<string> userEmails, Action<string> logAction = null)
+        {
+            userEmails = new List<string>();
+
             // By checking appidacr claim, we can know if the call was made by a user or by the system.
             // https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens
             string appidacr = claims.FirstOrDefault(claim => claim.Type == "appidacr")?.Value;
 
-            switch(appidacr)
+            switch (appidacr)
             {
-                case "0" :
-                    Claim[] claimsArray = claims.ToArray();
-                    userEmails = PossibleEmailClaimTypes
-                        .Select(claimType => claimsArray.FirstOrDefault(claim => claim.Type == claimType))
-                        .Where(claim => claim != null)
-                        .Select(claim => claim.Value)
-                        .Distinct()
-                        .ToList();
-                    return userEmails.Any();
-
-                case "2" :
+                case "0":
+                    return GetUserDetailsFromClaims(claims, out userEmails);
+
+                case "2":
                     return true;
 
-                default :
+                default:
+                    logAction?.Invoke("this is neither a valid call from user nor a valid server to server call");
```

Comment:
We haven't checked for `logAction == null` on line 18  as well. Should we be doing it there also?

Reply:
Yes we should, thanks for pointing that out.